File: PetriNetPlugin/src/org/workcraft/plugins/petri/tools/SimulationModel.java
package org.workcraft.plugins.petri.tools;

public interface SimulationModel<Event, State> {
	boolean canFire(Event event);
	void fire(Event event);
	boolean canUnfire(Event event);
	void unfire(Event event);
	
	State saveState();
	void loadState(State state);
}

File: ScalaGraphEditorUtil/src/main/scala/org/workcraft/plugins/petri/tools/SimulationModel.scala
package org.workcraft.plugins.petri.tools

trait SimulationModel[M[_], Event, State] {
  def canFire(event : Event) : M[Boolean]
  def fire(event : Event) : M[Unit]
  def canUnfire(event : Event) : M[Boolean]
  def unfire(event : Event) : M[Unit]
  def saveState : M[State]
  def loadState(state : State) : M[Unit]
}
